managed/CounterStrikeSharp.API/Utilities.cs

```diff
@@ -103,5 +103,36 @@ public static unsafe string ReadStringUtf8(IntPtr ptr)
                 return Encoding.UTF8.GetString(buffer);
             }
         }
+        
+        public static IEnumerable<CCSPlayerController> FindTarget(CCSPlayerController? player, string pattern) {
+            var emptyResult = Enumerable.Empty<CCSPlayerController>();
+            var playerEntities = Utilities.FindAllEntitiesByDesignerName<CCSPlayerController>("cs_player_controller");
+            
+            if (pattern.StartsWith("#")) 
+            {
+                if (!int.TryParse(pattern.Substring(1), out var userId)) 
+                    return emptyResult;
+                
+                var playerController = playerEntities.Select(player => player).FirstOrDefault(player => player.UserId == userId);
```

Comment:
This select is unnecessary, and we should avoid shadowing the outer `player` variable.
```suggestion
                var playerController = playerEntities.FirstOrDefault(p =>p.UserId == userId);
```